org.mixedrealitytoolkit.input/Tracking/HandPoseDriver.cs

```diff
@@ -0,0 +1,267 @@
+
+        #region Private Functions
+        /// <summary>
+        /// Check the tracking state here to account for a bound but untracked interaction profile.
+        /// This could show up on runtimes where a controller is disconnected, hand tracking spins up,
+        /// but the interaction profile is not cleared. This is allowed, per-spec: "The runtime may
+        /// return the last-known interaction profile in the event that no controllers are active."
+        /// </summary>
+        private bool IsTrackingNone()
+        {
+            return m_trackingState == InputTrackingState.None;
+        }
+
+        /// <summary>
+        /// Workaround for missing device pose on devices without interaction profiles
+        /// for hands, such as Varjo and Quest. Should be removed once we have universal
+        /// hand interaction profile(s) across vendors.
+        /// </summary>
+        private bool TryGetPolyfillDevicePose(out Pose devicePose)
+        {
+            bool poseRetrieved = false;
+            Handedness handedness = HandNode.ToHandedness();
+
+            // palmPose retrieved in global space.
+            if (XRSubsystemHelpers.HandsAggregator != null &&
+                XRSubsystemHelpers.HandsAggregator.TryGetJoint(TrackedHandJoint.Palm, HandNode, out HandJointPose palmPose))
+            {
+                // XRControllers work in OpenXR scene-origin-space, so we need to transform
+                // our global palm pose back into scene-origin-space.
+                devicePose = PlayspaceUtilities.InverseTransformPose(palmPose.Pose);
+
+                switch (handedness)
+                {
+                    case Handedness.Left:
+                        devicePose.rotation *= leftPalmOffset;
+                        poseRetrieved = true;
+                        break;
+                    case Handedness.Right:
+                        devicePose.rotation *= rightPalmOffset;
+                        poseRetrieved = true;
+                        break;
+                    default:
+                        Debug.LogError("No polyfill available for device with handedness " + handedness);
+                        devicePose = Pose.identity;
+                        poseRetrieved = false;
+                        break;
+                };
+            }
+            else
+            {
+                devicePose = Pose.identity;
+            }
+
+            return poseRetrieved;
+        }
+
+        /// <summary>
+        /// Sets the transform that is being driven by the <see cref="TrackedPoseDriver"/>. This will only set requested values, but regardless of tracking state.
+        /// </summary>
+        /// <param name="newPosition">The new local position to possibly set.</param>
+        /// <param name="newRotation">The new local rotation to possibly set.</param>
+        protected virtual void ForceSetLocalTransform(Vector3 newPosition, Quaternion newRotation)
```

Comment:
I don't see this method used currently. Is that coming in a future PR, or is this stale?

Reply:
doh...This should have been called at line 76.... `SetLocalTransform`   is a base class, and won't apply pose updates if device isn't tracked.